src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -834,6 +865,20 @@ class module_sram : public module_impl
 
       // copy instruction into bo
       fill_ctrlpkt_buf(m_ctrlpkt_buf, data);
+
+#ifdef _DEBUG
+      if (xrt_core::config::get_feature_toggle(Debug_Bo_From_Elf_Feature)) {
+          std::string filename = "ctrlpktBo.bin";
+          std::ofstream ofs(filename, std::ios::out | std::ios::binary);
+
+          if (!ofs.is_open())
+            throw std::runtime_error("Failure opening file " + filename + " for writing!");
+
+          auto buf = m_ctrlpkt_buf.map<char*>();
+          ofs.write(buf, m_ctrlpkt_buf.size());
+      }
```

Comment:
This appears to be a repeated code block from above; I didn't notice that before.  If so, please refactor in a separate function that take output filename and xrt::bo to dump.  E.g.
dump_bo(m_ctrl_buf, "ctrlpktbo.bin");